### ISSUE TITLE
How to get a Selenide driver handler in a DockerBrowser test?

### ISSUE THREAD

**Issue by djangofan** [bonigarcia/selenium-jupiter#78]
Perhaps "Feature Request" ?

**Description of the problem**: 
How can I  get a Selenide driver handler in a DockerBrowser test?

This is what I tried, which seemed somewhat logical, but it didn't work for me.  Is there a way to do this?   I have Selenide tests in Selenium-Jupiter that I now want to dockerize without having to rewrite everything.

If there is an example somewhere, just point me to it.

**Browser and version**:   For this question, just Chrome or Firefox

**Operating system**:  Mac OSX

**Selenium-Jupiter version**:  latest

**Selenium-Jupiter use**: 

```
    @Test
    public void testContent(@DockerBrowser(type = FIREFOX, volumes = {"~:/home/selenium" }) RemoteWebDriver driver) {
        WebDriverRunner.setWebDriver(driver);
        SelenideDriver sd = (SelenideDriver)WebDriverRunner.getAndCheckWebDriver();
        sd.open("http://google.com");
```

**Selenium-Jupiter traces**: 

20:09:21.093 [Test worker] WARN io.github.bonigarcia.seljup.SeleniumExtension - Exception creating class io.github.bonigarcia.seljup.handler.RemoteDriverHandler
java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)

**Error trace**:

**Comment by djangofan** [comment]
I think I found it.  Sound correct?
https://github.com/bonigarcia/selenium-jupiter/blob/50e699cb3a8ef15474c8b80d787843aac2eab987/src/test/java/io/github/bonigarcia/seljup/test/selenide/SelenideDockerJupiterTest.java#L49

**Comment by djangofan** [comment]
Gettin close,  now I get:

```
21:05:15.146 [Test worker] WARN io.github.bonigarcia.seljup.SeleniumExtension - Exception creating class io.github.bonigarcia.seljup.handler.SelenideDriverHandler
java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
	at java.base/java.lang.Integer.valueOf(Integer.java:983)
	at io.github.bonigarcia.seljup.config.Config.parse(Config.java:290)
	at io.github.bonigarcia.seljup.config.Config.resolve(Config.java:281)
	at io.github.bonigarcia.seljup.config.Config.getDockerWaitTimeoutSec(Config.java:807)
	at io.github.bonigarcia.seljup.DockerService.<init>(DockerService.java:71)
```

**Comment by djangofan** [comment]
I am now trying your selenium-jupiter-examples project to see if I can get it working there...

**Comment by bonigarcia** [comment]
It seems there is a bug here. I have just executed that test within selenium-jupiter-examples. In Linux everything is fine:
```
boni@ubuntu:~/dev/selenium-jupiter-examples$ mvn test -Dtest=DockerSelenideTest
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------< io.github.bonigarcia:selenium-jupiter-examples >-----------
[INFO] Building selenium-jupiter-examples 3.3.4
[INFO] --------------------------------[ jar ]---------------------------------

...
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.github.bonigarcia.seljup.test.docker.DockerSelenideTest
2020-05-15 12:27:05 [main] INFO  i.g.b.seljup.DockerHubService.listTags(61) -- Getting browser image list from Docker Hub
2020-05-15 12:27:06 [main] INFO  i.g.b.s.handler.DockerDriverHandler.startDockerBrowser(719) -- Using OPERA version 67.0 (latest)
2020-05-15 12:27:06 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image selenoid/vnc:opera_67.0
2020-05-15 12:28:23 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image aerokube/selenoid:1.9.3
2020-05-15 12:28:25 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image selenoid/video-recorder:latest-release
2020-05-15 12:28:30 [main] INFO  i.g.bonigarcia.seljup.DockerService.startContainer(106) -- Starting Docker container aerokube/selenoid:1.9.3
2020-05-15 12:28:34 [main] DEBUG i.g.b.seljup.WebDriverCreator.createRemoteWebDriver(66) -- Creating WebDriver for OPERABLINK at http://172.17.0.1:32768/wd/hub with Capabilities {browserName: operablink, enableVNC: true, enableVideo: true, operaOptions: {args: [], extensions: []}, platform: ANY, screenResolution: 1920x1080x24, version: , videoFrameRate: 12, videoScreenSize: 1920x1080}
May 15, 2020 12:28:36 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Detected dialect: OSS
2020-05-15 12:28:36 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image psharkey/novnc:3.3-t6
2020-05-15 12:28:48 [main] INFO  i.g.bonigarcia.seljup.DockerService.startContainer(106) -- Starting Docker container psharkey/novnc:3.3-t6
2020-05-15 12:28:49 [main] INFO  i.g.b.s.handler.DockerDriverHandler.logSessionId(350) -- Session id 3f33ca654168aa17a54ec8cb38ca3bba
2020-05-15 12:28:49 [main] INFO  i.g.b.s.handler.DockerDriverHandler.logNoVncUrl(354) -- VNC URL (copy and paste in a browser navigation bar to interact with remote session)
2020-05-15 12:28:49 [main] INFO  i.g.b.s.handler.DockerDriverHandler.logNoVncUrl(356) -- http://172.17.0.1:32769/vnc.html?host=172.17.0.1&port=32768&path=vnc/3f33ca654168aa17a54ec8cb38ca3bba&resize=scale&autoconnect=true&password=selenoid
2020-05-15 12:29:16 [main] DEBUG i.g.b.s.handler.DockerDriverHandler.waitForRecording(1151) -- Waiting for recording to be available
2020-05-15 12:29:18 [pool-6-thread-1] INFO  i.g.bonigarcia.seljup.DockerService.stopAndRemoveContainer(231) -- Stopping Docker container aerokube/selenoid:1.9.3
2020-05-15 12:29:18 [pool-6-thread-2] INFO  i.g.bonigarcia.seljup.DockerService.stopAndRemoveContainer(231) -- Stopping Docker container psharkey/novnc:3.3-t6
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 136.04 s - in io.github.bonigarcia.seljup.test.docker.DockerSelenideTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:22 min
[INFO] Finished at: 2020-05-15T12:29:20+02:00
[INFO] ------------------------------------------------------------------------
```
... but it Windows (with Docker Desktop) the same test is not working:
```
C:\Users\boni\dev\selenium-jupiter-examples>mvn test -Dtest=DockerSelenideTest
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building selenium-jupiter-examples 3.3.4
[INFO] ------------------------------------------------------------------------

...
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.github.bonigarcia.seljup.test.docker.DockerSelenideTest
2020-05-15 10:37:50 [main] INFO  i.g.b.s.handler.DockerDriverHandler.startDockerBrowser(719) -- Using OPERA version 67.0 (latest)
2020-05-15 10:37:50 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image selenoid/vnc:opera_67.0
2020-05-15 10:37:53 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image aerokube/selenoid:1.9.3
2020-05-15 10:37:56 [main] INFO  i.g.bonigarcia.seljup.DockerService.pullImage(200) -- Pulling Docker image selenoid/video-recorder:latest-release
2020-05-15 10:37:58 [main] INFO  i.g.bonigarcia.seljup.DockerService.startContainer(106) -- Starting Docker container aerokube/selenoid:1.9.3
2020-05-15 10:37:58 [pool-5-thread-1] INFO  i.g.bonigarcia.seljup.DockerService.stopAndRemoveContainer(231) -- Stopping Docker container aerokube/selenoid:1.9.3
2020-05-15 10:37:58 [pool-5-thread-1] WARN  i.g.bonigarcia.seljup.DockerService.stopAndRemoveContainer(236) -- Exception stopping container aerokube/selenoid:1.9.3
java.lang.NullPointerException: path is 'null'.
        at jersey.repackaged.com.google.common.base.Preconditions.checkNotNull(Preconditions.java:226)
        at org.glassfish.jersey.client.JerseyWebTarget.path(JerseyWebTarget.java:151)
        at org.glassfish.jersey.client.JerseyWebTarget.path(JerseyWebTarget.java:59)
        at com.spotify.docker.client.DefaultDockerClient.stopContainer(DefaultDockerClient.java:855)
        at io.github.bonigarcia.seljup.DockerService.stopContainer(DockerService.java:249)
        at io.github.bonigarcia.seljup.DockerService.stopAndRemoveContainer(DockerService.java:233)
        at io.github.bonigarcia.seljup.handler.DockerDriverHandler.lambda$stopContainers$0(DockerDriverHandler.java:582)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 9.215 s <<< FAILURE! - in io.github.bonigarcia.seljup.test.docker.DockerSelenideTest
[ERROR] testDockerSelenide{SelenideDriver}  Time elapsed: 9.206 s  <<< ERROR!
org.junit.jupiter.api.extension.ParameterResolutionException: Failed to resolve parameter [com.codeborne.selenide.SelenideDriver arg0] in method [void io.github.bonigarcia.seljup.test.docker.DockerSelenideTest.testDockerSelenide(com.codeborne.selenide.SelenideDriver) throws java.lang.InterruptedException]: io.github.bonigarcia.seljup.SeleniumJupiterException: Exception resolving driver in Docker (OPERA )
Caused by: io.github.bonigarcia.seljup.SeleniumJupiterException: io.github.bonigarcia.seljup.SeleniumJupiterException: Exception resolving driver in Docker (OPERA )
Caused by: io.github.bonigarcia.seljup.SeleniumJupiterException: Exception resolving driver in Docker (OPERA )
Caused by: com.spotify.docker.client.exceptions.DockerRequestException:
Request error: POST npipe://localhost:80/containers/create: 500, body: /c\Users\boni\dev\selenium-jupiter-examples\.%!(EXTRA string=is not a valid Windows path)

Caused by: javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   DockerSelenideTest.testDockerSelenide » ParameterResolution Failed to resolve ...
[INFO]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 13.442 s
[INFO] Finished at: 2020-05-15T10:37:59+02:00
[INFO] Final Memory: 16M/243M
[INFO] ------------------------------------------------------------------------
```
I suppose in Mac it's happening something wrong too.

I plan to implement several new features in Selenium-Jupiter during the next months. I'll fix this one for the next version.

**Comment by djangofan** [comment]
@bonigarcia   When I tried your **selenium-jupiter-examples**  I did manage to get it working on my Mac, such as the recording example.    It is just that when I tried to create a project from scratch, whatever it was that I did was causing that problem.  The error is obscure enough that I don't know what I did wrong.   So, I will proceed by starting with your template and going from there.

I agree there is an issue there.   I just couldn't say if I just configured it wrong or if it is a real bug.   I'll build out a project step-by-step and if I run into the issue again, hopefully I will know what I did.

**Comment by bonigarcia** [comment]
I don't know exactly why, but it seems this issue is fixed in the master branch of Selenium-Jupiter. I tried again the same test in the project selenium-jupiter-examples, but using Selenium-Jupiter `3.3.5-SNAPSHOT` (installed from the source code) in a Windows machine. The test is passing now:

```
C:\Users\boni\dev\selenium-jupiter-examples>mvn test -Dtest=DockerSelenideTest
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building selenium-jupiter-examples 3.3.4
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ selenium-jupiter-examples ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\boni\dev\selenium-jupiter-examples\src\main\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ selenium-jupiter-examples ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ selenium-jupiter-examples ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ selenium-jupiter-examples ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ selenium-jupiter-examples ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.github.bonigarcia.seljup.test.docker.DockerSelenideTest
2020-09-09 18:32:45 [main] INFO  i.g.b.s.handler.DockerDriverHandler.startDockerBrowser(720) -- Using OPERA version 70.0 (latest)
2020-09-09 18:32:45 [main] INFO  i.g.bonigarcia.seljup.DockerService.startContainer(118) -- Starting Docker container aerokube/selenoid:1.9.3
2020-09-09 18:32:47 [main] DEBUG i.g.b.seljup.WebDriverCreator.createRemoteWebDriver(66) -- Creating WebDriver for OPERABLINK at http://localhost:32771/wd/hub with Capabilities {browserName: operablink, enableVNC: true, enableVideo: true, operaOptions: {args: [], extensions: []}, platform: ANY, screenResolution: 1920x1080x24, version: , videoFrameRate: 12, videoScreenSize: 1920x1080}
sep 09, 2020 6:32:48 PM org.openqa.selenium.remote.ProtocolHandshake createSession
INFORMACIÓN: Detected dialect: OSS
2020-09-09 18:32:48 [main] INFO  i.g.bonigarcia.seljup.DockerService.startContainer(118) -- Starting Docker container psharkey/novnc:3.3-t6
2020-09-09 18:32:49 [main] INFO  i.g.b.s.handler.DockerDriverHandler.logSessionId(351) -- Session id 8c576330a8df5518343420a65698b462
2020-09-09 18:32:49 [main] INFO  i.g.b.s.handler.DockerDriverHandler.logNoVncUrl(355) -- VNC URL (copy and paste in a browser navigation bar to interact with remote session)
2020-09-09 18:32:49 [main] INFO  i.g.b.s.handler.DockerDriverHandler.logNoVncUrl(357) -- http://localhost:32772/vnc.html?host=localhost&port=32771&path=vnc/8c576330a8df5518343420a65698b462&resize=scale&autoconnect=true&password=selenoid
2020-09-09 18:33:10 [main] DEBUG i.g.b.s.handler.DockerDriverHandler.waitForRecording(1173) -- Waiting for recording to be available
2020-09-09 18:33:11 [pool-2-thread-1] INFO  i.g.bonigarcia.seljup.DockerService.stopAndRemoveContainer(243) -- Stopping Docker container aerokube/selenoid:1.9.3
2020-09-09 18:33:11 [pool-2-thread-2] INFO  i.g.bonigarcia.seljup.DockerService.stopAndRemoveContainer(243) -- Stopping Docker container psharkey/novnc:3.3-t6
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 28.498 s - in io.github.bonigarcia.seljup.test.docker.DockerSelenideTest
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 32.549 s
[INFO] Finished at: 2020-09-09T18:33:13+02:00
[INFO] Final Memory: 20M/434M
[INFO] ------------------------------------------------------------------------
```


@djangofan Could you please try your code in Mac with Selenium-Jupiter `3.3.5-SNAPSHOT`?

**Comment by bonigarcia** [comment]
This should be fixed in Selenium-Jupiter 3.3.5, just released.
